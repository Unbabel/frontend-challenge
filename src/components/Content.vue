<template>
    <section class="content">
        <Transition>
            <TransitionGroup v-if="!listIsEmpty" tag="ul" class="list">
                <li v-for="note in notes" :key="note.id">
                    <Note :data="note" />
                </li>
            </TransitionGroup>
        </Transition>
        <button @click="addNote" class="btn" :aria-label="$t('aria.add')">
            <Icon name="add" />
        </button>
    </section>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { mapMutations, mapState } from 'vuex'
import Note from '@/components/Note.vue'
import Icon from '@/components/icons/Icon.vue'

export default defineComponent({
    components: {
        Note,
        Icon
    },
    computed: {
        ...mapState(['notes']),
        listIsEmpty() {
            return this.notes.length === 0
        }
    },
    methods: {
        ...mapMutations(['addNote'])
    }
})
</script>

<style src="@/assets/css/transitions.scss" lang="scss" scoped />
<style lang="scss" scoped>
$header: 69px;
$footer: 36px;

.content {
    min-height: calc(100% - $header - $footer);
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
}

.list {
    width: 100%;
    list-style: none;
    padding: 0;
    margin: 0;

    @media(min-width: 768px) {
        margin-top: 24px;
        max-width: 738px;
    }
}

.btn {
    margin: 24px 0;
}
</style>