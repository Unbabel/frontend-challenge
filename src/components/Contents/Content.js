import * as React from 'react'

const Content = ({ value, inputType, onChange, onBlur, placeholder }) => {
    return (inputType === "single" ? < input value={value}
        onChange={onChange}
        onBlur={onBlur}
        autoFocus={true}
        placeholder={placeholder}
    />
        :
        <textarea value={value}
            onChange={onChange}
            onBlur={onBlur}
            autoFocus={true}
            placeholder={placeholder}
        />
    )
}
export default Content;