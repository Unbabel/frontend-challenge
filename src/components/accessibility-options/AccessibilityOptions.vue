<script>
	// Store
	import { mapState, mapActions } from "vuex";

	// Components
	import Modal from "../modal/Modal.vue";
	import ToggleButton from "../toggle-button/ToggleButton.vue";

	// Utils
	import { addOrRemoveClassNameFromElTag } from "../../shared/utils/accessibilityOptions";

	export default {
		components: {
			Modal,
			ToggleButton,
		},
		methods: {
			...mapActions({
				toggleModalVisibility: "toggleModalVisibility",
				toggleAccessibilityOption: "toggleAccessibilityOption",
			}),
			openModal() {
				this.toggleModalVisibility();
			},
			toggleElderMode() {
				addOrRemoveClassNameFromElTag("html", "elder");
				this.toggleAccessibilityOption("elder");
			},
			toggleColorBlindness() {
				this.toggleAccessibilityOption("colorBlindness");
			},
		},
		computed: {
			...mapState({
				showModal: "showModal",
				isElderModeActive: (state) => state.accessibilityOptions.elder,
				isColorBlindnessActive: (state) =>
					state.accessibilityOptions.colorBlindness,
			}),
		},
	};
</script>

<template>
	<div>
		<div class="accessibility-container" aria-label="Accessibility Options">
			<div
				class="accessibility"
				@click="openModal"
				data-test="accessibility-div"
			>
				<img
					src="/src/assets/accessibility.svg"
					alt="accessibility-icon"
					data-test="accessibility-icon"
				/>
			</div>
		</div>
		<Modal>
			<h3>Accessibility Profiles</h3>
			<div class="accessibility-options-grid">
				<p>Elder</p>
				<ToggleButton
					:onChangeCallback="toggleElderMode"
					:checked="isElderModeActive"
				/>
				<p>Color Blindness</p>
				<ToggleButton
					:onChangeCallback="toggleColorBlindness"
					:checked="isColorBlindnessActive"
				/>
			</div>
		</Modal>
	</div>
</template>

<style scoped>
	.accessibility-container {
		position: fixed;
		z-index: 10;
		bottom: 1.25rem;
		left: 0.625rem;
		width: fit-content;
	}
	.accessibility {
		background: var(--palette-purple);
		border-radius: 50%;
		margin: 0.625rem;
		height: fit-content;
		width: fit-content;
		padding: 0.5rem;
		cursor: pointer;

		box-shadow: 0 0 0 0 rgb(129, 94, 194);
		transform: scale(1);
		animation: pulse 2s infinite;
	}

	img {
		height: 2rem;
	}

	@keyframes pulse {
		0% {
			transform: scale(0.95);
			box-shadow: 0 0 0 0 rgba(129, 94, 194, 0.7);
		}

		70% {
			transform: scale(1);
			box-shadow: 0 0 0 10px rgba(142, 68, 173, 0);
		}

		100% {
			transform: scale(0.95);
			box-shadow: 0 0 0 0 rgba(142, 68, 173, 0);
		}
	}

	h3 {
		text-align: center;
	}

	.accessibility-option {
		column-gap: 2rem;
	}

	.accessibility-options-grid {
		display: grid;
		grid-template-columns: 70% 29%;
		align-items: center;
	}
</style>
