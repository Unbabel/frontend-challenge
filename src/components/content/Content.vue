<script setup></script>

<template>
  <div>
    <slot />
  </div>
</template>

<style scoped></style>
