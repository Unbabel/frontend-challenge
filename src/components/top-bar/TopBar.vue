<script setup>
	// Components
	import PageTitle from "../page-title/PageTitle.vue";
	import IconButton from "../icon-button/IconButton.vue";
</script>

<template>
	<div class="top-bar-container">
		<div class="top-bar-inner-container flex-center flex-center-row">
			<PageTitle text="Transcriptions" />
			<div
				class="top-bar-actions-container flex-center flex-center-row"
				data-test="top-bar-actions"
			>
				<IconButton
					v-for="action in $route.meta?.actions"
					@onClickCallback="action.callback"
					:icon="action.icon"
					:key="action.id"
				/>
			</div>
		</div>
	</div>
</template>

<style scoped>
	.top-bar-container {
		width: 100%;
		height: 4.3rem;
		background-color: var(--palette-white);
		box-shadow: 0 1px 3px 0 var(--palette-transp-black);
	}

	.top-bar-inner-container {
		height: 100%;
		width: 55%;
		margin: 0 auto;
		justify-content: space-between;
	}

	.top-bar-actions-container {
		column-gap: 0.5rem;
	}

	@media (max-width: 768px) {
		.top-bar-inner-container {
			width: 90%;
		}
	}
</style>
