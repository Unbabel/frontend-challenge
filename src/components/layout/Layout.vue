<script setup>
	//  Components
	import TopBar from "../top-bar/TopBar.vue";
	import Content from "../content/Content.vue";
	import AccessibilityOptions from "../accessibility-options/AccessibilityOptions.vue";
</script>

<template>
	<div class="layout-container">
		<TopBar />
		<Content>
			<slot />
		</Content>
		<AccessibilityOptions />
	</div>
</template>

<style scoped></style>
