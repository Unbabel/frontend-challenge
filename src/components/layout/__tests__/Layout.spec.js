import Vuex from "vuex";
import { describe, it, expect, beforeEach, vi } from "vitest";
import { mount, createLocalVue, shallowMount } from "@vue/test-utils";

// Components
import Layout from "../Layout.vue";
import TopBar from "../../top-bar/TopBar.vue";
import Content from "../../content/Content.vue";
import AccessibilityOptions from "../../accessibility-options/AccessibilityOptions.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

const $route = {
	fullPath: "full/path",
};

const store = new Vuex.Store({
	state: {
		showModal: true,
		accessibilityOptions: {
			elder: false,
		},
	},
});

describe("Layout", () => {
	let wrapper;

	beforeEach(() => {
		// Arrange
		wrapper = shallowMount(Layout, {
			localVue,
			store,
			mocks: {
				$route,
			},
		});
	});

	it("matches snapshot", () => {
		// Assert
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("renders properly", () => {
		// Act
		const topbar = wrapper.findComponent(TopBar);
		const content = wrapper.findComponent(Content);
		const accessibilityOptions =
			wrapper.findComponent(AccessibilityOptions);

		// Assert
		expect(topbar.exists()).toBe(true);
		expect(content.exists()).toBe(true);
		expect(accessibilityOptions.exists()).toBe(true);
	});
});
