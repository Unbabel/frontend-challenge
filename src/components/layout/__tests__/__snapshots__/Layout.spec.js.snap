// Vitest Snapshot v1

exports[`Layout > matches snapshot 1`] = `
"<div data-v-c9bfb757=\\"\\" class=\\"layout-container\\">
  <anonymous-stub data-v-c9bfb757=\\"\\"></anonymous-stub>
  <anonymous-stub data-v-c9bfb757=\\"\\"></anonymous-stub>
  <anonymous-stub data-v-c9bfb757=\\"\\"></anonymous-stub>
</div>"
`;
